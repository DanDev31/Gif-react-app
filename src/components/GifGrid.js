import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const {data: images, loading} = useFetchGifs( category );

    console.log(category)
 
    return (
        <div>
           <h2 className="category__title">{ category }</h2> 

            { loading && <p>Loading...</p> }

            <div className = "gif__container">
              {
              images.map( img => 
                  <GifGridItem 
                        key={ img.id }
                            { ...img }
                  /> 
              )
              }
              
             </div> 
            
        </div>
    )
}
