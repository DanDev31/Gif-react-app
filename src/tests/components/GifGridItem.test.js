import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en GifGridItem.js', () =>{

    test('Debe mostrar el componente correctamente', () =>{

        const wrapper = shallow(<GifGridItem/>)
        expect( wrapper ).toMatchSnapshot();
    })
    
})