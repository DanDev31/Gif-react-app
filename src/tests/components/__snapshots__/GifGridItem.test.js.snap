// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js Debe mostrar el componente correctamente 1`] = `
<div
  className="gif__card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
